src/chunk.c

```diff
@@ -203,6 +203,24 @@ void chunkqueue_reset(chunkqueue *cq) {
 	cq->bytes_out = 0;
 }
 
+void chunkqueue_append_file_fd(chunkqueue *cq, buffer *fn, int fd, off_t offset, off_t len) {
+	chunk *c;
+
+	if (0 == len) return;
```

Comment:
either `fd` leaks here or it leaks later - define and document whether `chunkqueue_append_file_fd` is supposed to own the passed `fd` and handle it accordingly.